[docs] Expo CLI: Improve the docs and add note for creating a new expo project

amandeepmittal opened the pull request:
# Why

<!--
Please describe the motivation for this PR, and link to relevant GitHub issues, forums posts, or feature requests.
-->

Refs ENG-5549 & ENG-5387

# How

<!--
How did you build this feature or fix this bug and why?
-->

This PR:
- adds a note to indicate that about creating a new Expo project using `npx create-expo-app`
- Updates inconsistent verbiage

# Test Plan

<!--
Please describe how you tested this change and how a reviewer could reproduce your test, especially if this PR does not include automated tests! If possible, please also provide terminal output and/or screenshots demonstrating your test/reproduction.
-->

The changes have been tested locally.

# Checklist

<!--
Please check the appropriate items below if they apply to your diff. This is required for changes to Expo modules.
-->

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [ ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

amandeepmittal committed:
[docs] Expo CLI: Improve the docs

jonsamp commented on docs/pages/workflow/expo-cli.md:
```diff
@@ -5,30 +5,41 @@ maxHeadingDepth: 4
-- [Building binaries](/distribution/building-standalone-apps.md) (`aab` and `ipa` files) to be [uploaded to the App Store and Play Store](/distribution/uploading-apps.md)
+- To run a project's [development server](/guides/how-expo-works/#expo-development-server), view logs, open the app in a simulator, etc., during the development phase of an app
+- [Publishing](publishing) an app's JavaScript and other assets and managing to release them as updates
+- [Building binaries](/distribution/building-standalone-apps) (`.aab` and `.ipa` files) and [uploading them to app stores](/distribution/uploading-apps)
```
```suggestion
- [Building binaries](/distribution/building-standalone-apps) (`.aab`, `.apk`, and `.ipa` files) and [uploading them to app stores](/distribution/uploading-apps)
```

jonsamp commented:
@EvanBacon can you review this and make sure it makes sense with versioned Expo CLI plans/work?

amandeepmittal committed:
Update docs/pages/workflow/expo-cli.md

Co-authored-by: Jon Samp <sampjon@gmail.com>

EvanBacon commented on docs/pages/workflow/expo-cli.md:
```diff
@@ -5,30 +5,41 @@ maxHeadingDepth: 4
-- [Building binaries](/distribution/building-standalone-apps.md) (`aab` and `ipa` files) to be [uploaded to the App Store and Play Store](/distribution/uploading-apps.md)
+- To run a project's [development server](/guides/how-expo-works/#expo-development-server), view logs, open the app in a simulator, etc., during the development phase of an app
+- [Publishing](publishing) an app's JavaScript and other assets and managing to release them as updates
+- [Building binaries](/distribution/building-standalone-apps) (`.aab`, `.apk`, and `.ipa` files) and [uploading them to app stores](/distribution/uploading-apps)
```
Even though Expo CLI is technically capable of building binaries in the cloud, we should only reference EAS Build/Submit.

amandeepmittal replied:
Maybe, we should remove the last point about building binaries?

If not, the other way I can think of is to adjust the text like this:

```markdown
- [Building binaries](/distribution/building-standalone-apps) (`.aab`, `.apk`, and `.ipa` files) and [uploading them to app stores](/distribution/uploading-apps) using Classic Build Service. Although, it is recommended to use EAS Build to build binaries and submit them to app stores with support for custom native code and smaller app archives.
```

amandeepmittal committed:
Remove Building binaries point

amandeepmittal commented:
Removed the "Building binaries" and "uploading them to app stores" bullet point in favor of EAS. I think removing the point will not lead to confusion. Also, when referencing the commands later in this page under [Deprecated section](https://docs.expo.dev/workflow/expo-cli/#deprecated) it also says the same.

Added another point about logging into Expo account. @jonsamp, lmk if you think this point is not worth highlighting as list point.

amandeepmittal commented on docs/pages/workflow/expo-cli.md:
```diff
@@ -5,30 +5,41 @@ maxHeadingDepth: 4
 import { Terminal } from '~/ui/components/Snippet';
 
-Expo CLI is a command line app that is the main interface between a developer and Expo tools. You'll use it for a variety of tasks, such as:
+Expo CLI is the primary command-line interface between a developer and Expo tools. It is used for a variety of tasks, such as:
```
I'll remove the "primary" interface part.

amandeepmittal committed:
Remove primary/main interface part

jonsamp commented:
> Removed the "Building binaries" and "uploading them to app stores" bullet point in favor of EAS. I think removing the point will not lead to confusion. Also, when referencing the commands later in this page under [Deprecated section](https://docs.expo.dev/workflow/expo-cli/#deprecated) it also says the same.
> 
> Added another point about logging into Expo account. @jonsamp, lmk if you think this point is not worth highlighting as list point.

I like that you added a bullet about logging in!
